api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -208,18 +208,17 @@ private Workspace createWorkspaceImpl(Workspace workspace) {
     validateWorkspaceApiModel(workspace);
 
     DbUser user = userProvider.get();
-    String workspaceNamespace;
-    DbBillingProjectBufferEntry bufferedBillingProject;
+    final DbBillingProjectBufferEntry bufferedBillingProject;
     try {
-      bufferedBillingProject = billingProjectBufferService.assignBillingProject(user);
+      bufferedBillingProject = billingProjectBufferService.assignBillingProject(user, workspace);
     } catch (EmptyBufferException e) {
       throw new TooManyRequestsException();
```

Comment:
Do we need to put the original exception inside? i.e.` TooManyRequestsException(e)`

Reply:
sounds like a good idea